VocaDbModel/DataContracts/ReleaseEvents/ReleaseEventSeriesForApiContract.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Linq;
+using System.Runtime.Serialization;
+using VocaDb.Model.Domain;
+using VocaDb.Model.Domain.Globalization;
+using VocaDb.Model.Domain.Images;
+using VocaDb.Model.Domain.ReleaseEvents;
+
+namespace VocaDb.Model.DataContracts.ReleaseEvents {
+
+	public class ReleaseEventSeriesForApiContract {
+
+		public ReleaseEventSeriesForApiContract() { }
+
+		public ReleaseEventSeriesForApiContract(ReleaseEventSeries series, ContentLanguagePreference languagePreference, ReleaseEventSeriesOptionalFields fields, IEntryThumbPersister thumbPersister) {
+
+			Category = series.Category;
+			Id = series.Id;
+			Name = series.TranslatedName[languagePreference];
+			Status = series.Status;
+			Version = series.Version;
+
+			if (fields.HasFlag(ReleaseEventSeriesOptionalFields.AdditionalNames)) {
+				AdditionalNames = series.Names.GetAdditionalNamesStringForLanguage(languagePreference);
+			}
+
+			if (fields.HasFlag(ReleaseEventSeriesOptionalFields.Description)) {
+				Description = series.Description;
+			}
+
+			if (fields.HasFlag(ReleaseEventSeriesOptionalFields.Events)) {
+				Events = series.Events.Select(e => new ReleaseEventContract(e, languagePreference)).ToArray();
+			}
+
+			if (thumbPersister != null && fields.HasFlag(ReleaseEventSeriesOptionalFields.MainPicture)) {
+				MainPicture = EntryThumbForApiContract.Create(EntryThumb.Create(series), thumbPersister);
+			}
+
+			if (fields.HasFlag(ReleaseEventSeriesOptionalFields.Names)) {
+				Names = series.Names.Select(n => new LocalizedStringContract(n)).ToArray();
+			}
+
+			if (fields.HasFlag(ReleaseEventSeriesOptionalFields.WebLinks)) {
+				WebLinks = series.WebLinks.Select(w => new WebLinkForApiContract(w)).ToArray();
+			}
+
+		}
+
+		/// <summary>
+		/// Comma-separated list of all other names that aren't the display name.
+		/// </summary>
+		[DataMember(EmitDefaultValue = false)]
+		public string AdditionalNames { get; set; }
+
+		[DataMember]
+		public EventCategory Category { get; set; }
+
+		[DataMember(EmitDefaultValue = false)]
+		public string Description { get; set; }
+
+		[DataMember]

```

Comment:
I would add [DataMember(EmitDefaultValue = false)] to all optional fields, including this one. WebLinks too, I think. It's better to return no list than empty list when the field is not requested. I know for events the EmitDefaultValue isn't there, but maybe it should.

Reply:
And this field should be ReleaseEventForApiContract instead of ReleaseEventContract. The "non-API" versions are mainly for internal use.

Reply:
Thanks for your review!